public/app/features/alerting/unified/components/receivers/grafanaAppReceivers/grafanaApp.ts

```diff
@@ -0,0 +1,53 @@
+import { useGetSingle } from 'app/features/plugins/admin/state/hooks';
+import { CatalogPlugin } from 'app/features/plugins/admin/types';
+import { Receiver } from 'app/plugins/datasource/alertmanager/types';
+
+import { useGetOnCallIntegrations } from '../../../api/onCallApi';
+
+import { isOnCallReceiver } from './onCall/onCall';
+import { AmRouteReceiver, GrafanaAppReceiverEnum, GRAFANA_APP_PLUGGIN_IDS, ReceiverWithTypes } from './types';
+
+export const useGetAppIsInstalledAndEnabled = (grafanaAppType: GrafanaAppReceiverEnum) => {
+  const plugin: CatalogPlugin | undefined = useGetSingle(GRAFANA_APP_PLUGGIN_IDS[grafanaAppType]);
+  return plugin?.isInstalled && !plugin?.isDisabled && plugin?.isPublished && plugin?.type === 'app'; // fetches the plugin settings for this Grafana instance
```

Comment:
This comment kind of confuses me. Is it correct for this line? Maybe could it say it `checks` the plugin settings instead of `fetches`?

Reply:
True, this comment refers to the previous line.I'll move it.